node.gyp

```diff
@@ -704,6 +709,15 @@
                 'deps/http_parser/http_parser.gyp:http_parser'
               ]
             }],
+            [ 'node_shared_nghttp2=="false"', {
+              'dependencies': [
+                'deps/nghttp2/nghttp2.gyp:nghttp2'
+              ],
+              'defines': [
+                # We're using the nghttp2 static lib
+                'NGHTTP2_STATICLIB'
+              ]
```

Comment:
I'd move the define to the `direct_dependent_settings` block in `deps/nghttp2/nghttp2.gyp`.  I did a similar thing in #15487 just now but I'll rebase if this lands before that.